File: java/algorithms/quickSort.java
package algorithms;
import java.io.*;

class main{
    public static void quickSort(int[] arr, int start, int end){
        if(start >= end){
            return;
        }

        int povit = arr[end], l_end = start, l = start, r = end;

        while(l < r){
            if(arr[l] < povit){
                int temp = arr[l];
                arr[l] = arr[l_end];
                arr[l_end] = temp;
                l_end++;
            }

            l++;
        }
           
        arr[r] = arr[l_end];
        arr[l_end] = povit;
        
        quickSort(arr, start, l_end -1);
        quickSort(arr, l_end+1, end);

        return;
    }   
    
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String in[] = br.readLine().split(",");
        int arr[] = new int[in.length];

        for(int i = 0; i < in.length; i++){
            arr[i] = Integer.valueOf(in[i]);
        }

        quickSort(arr, 0, arr.length -1);

        for(int i : arr){
            System.out.print(i + ", ");
        }


        return;
    }
}

File: interview/c++/quickSort.cpp
#include <vector>
#include <iostream>

using namespace std;

void QuickSort(vector<int>& arr, int start, int end){
    if(start >= end){
        return;
    }

    int povit = arr[end], l_end = start;

    for(int i = start; i < end; i++){
        if(arr[i] < povit){
            int temp = arr[i];
            arr[i] = arr[l_end];
            arr[l_end] = temp;
            l_end++;
        }
    }

    arr[end] = arr[l_end];
    arr[l_end] = povit;

    QuickSort(arr, start, l_end -1);
    QuickSort(arr, l_end+1, end);

    return;
}


template<typename T> ostream &operator<<(ostream& str, vector<T> arr){
    for(T i : arr){
        str << i << ", ";
    }

    str << endl;

    return str;
}


int main(){
    vector<int> arr{1,8,4,1,3,5,8,12};

    QuickSort(arr, 0, arr.size() - 1);

    cout << arr << endl;

    return 0;
}